gulpfile.js

```diff
@@ -29,6 +27,7 @@
 	let angularComponents;
```

Comment:
Take a look at what I did for extensions, to check whether the file exists, and replace it with an empty array if it doesn't. I think that would be cleaner.